### ISSUE TITLE
Linter exited successfully after an error

### ISSUE THREAD

**Issue by locks** [ember-template-lint/ember-template-lint#2512]
Today while upgrading to ember-template-lint v4 I noticed something I thought was peculiar.
When running e-t-l with a `pending` property still in the configuration, an error is triggered but the command returns a successful code. This means that CI is incorrectly passing all the checks.

```
❯ flax run lint:hbs
  npm   ✓ yarn    pnpm
yarn run v1.22.11
$ ember-template-lint . --quiet
(node:12379) UnhandledPromiseRejectionWarning: Error: Unknown top-level configuration property detected: pending
    at validateRootProperties (file:///Users/ricardo/src/_auditboard/soxhub-client/node_modules/ember-template-lint/lib/get-config.js:146:13)
    at getProjectConfig (file:///Users/ricardo/src/_auditboard/soxhub-client/node_modules/ember-template-lint/lib/get-config.js:420:5)
    at async Linter.loadConfig (file:///Users/ricardo/src/_auditboard/soxhub-client/node_modules/ember-template-lint/lib/linter.js:45:19)
    at async Linter.verify (file:///Users/ricardo/src/_auditboard/soxhub-client/node_modules/ember-template-lint/lib/linter.js:301:7)
    at async run (file:///Users/ricardo/src/_auditboard/soxhub-client/node_modules/ember-template-lint/bin/ember-template-lint.js:221:21)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:12379) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:12379) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
✨  Done in 1.87s.
```

**Comment by bmish** [comment]
I tested this out in a brand-new Ember app and saw the correct exit code:

```
~/Development/ember-test-app main * yarn lint:hbs
yarn run v1.22.18
$ ember-template-lint .
ember-test-app/node_modules/ember-template-lint/lib/get-config.js:146
      throw new Error(`Unknown top-level configuration property detected: ${key}`);
            ^

Error: Unknown top-level configuration property detected: pending
    at validateRootProperties (ember-test-app/node_modules/ember-template-lint/lib/get-config.js:146:13)
    at getProjectConfig (ember-test-app/node_modules/ember-template-lint/lib/get-config.js:420:5)
    at async Linter.loadConfig (ember-test-app/node_modules/ember-template-lint/lib/linter.js:45:19)
    at async Linter.verify (ember-test-app/node_modules/ember-template-lint/lib/linter.js:301:7)
    at async run (ember-test-app/node_modules/ember-template-lint/bin/ember-template-lint.js:221:21)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

```
ember-cli: 4.3.0
node: 16.15.0
os: darwin arm64

ember-template-lint 4.8.0
```

Can you ensure you are on the latest version of ember-template-lint and let us know if anything else is unique about your setup (I see `flax` in your output?).

**Comment by joshuabremerdexcom** [comment]
I'm able to recreate this issue too. I believe the difference is that Node 16 now treats these as errors, and Node 14 does not. I was able to work around this with the following:

Recreation Steps:
```
nvm use 14
npx ember-cli new recreate-ember-template-lint--issue
cd recreate-ember-template-lint--issue
yarn add ember-template-lint
sed -i '' 's/recommended/octane/g' .template-lintrc.js
yarn run ember-template-lint .
echo $?
```

Results:

```
~/repos/ember-template-lint-recreation/recreate-ember-template-lint--issue main*
❯ yarn run ember-template-lint .

yarn run v1.22.19
$ /Users/jbremer/repos/ember-template-lint-recreation/recreate-ember-template-lint--issue/node_modules/.bin/ember-template-lint .
(node:68184) UnhandledPromiseRejectionWarning: Error: Cannot find configuration for extends: octane
    at processExtends (file:///Users/jbremer/repos/ember-template-lint-recreation/recreate-ember-template-lint--issue/node_modules/ember-template-lint/lib/get-config.js:277:15)
    at getProjectConfig (file:///Users/jbremer/repos/ember-template-lint-recreation/recreate-ember-template-lint--issue/node_modules/ember-template-lint/lib/get-config.js:425:5)
    at async Linter.loadConfig (file:///Users/jbremer/repos/ember-template-lint-recreation/recreate-ember-template-lint--issue/node_modules/ember-template-lint/lib/linter.js:46:19)
    at async Linter.verify (file:///Users/jbremer/repos/ember-template-lint-recreation/recreate-ember-template-lint--issue/node_modules/ember-template-lint/lib/linter.js:302:7)
    at async run (file:///Users/jbremer/repos/ember-template-lint-recreation/recreate-ember-template-lint--issue/node_modules/ember-template-lint/bin/ember-template-lint.js:221:21)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:68184) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:68184) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
✨  Done in 1.20s.

~/repos/ember-template-lint-recreation/recreate-ember-template-lint--issue main*
❯ echo $?

0
```

Version info:

```
~/repos/ember-template-lint-recreation/recreate-ember-template-lint--issue main*
❯ yarn run ember --version
yarn run v1.22.19
$ /Users/jbremer/repos/ember-template-lint-recreation/recreate-ember-template-lint--issue/node_modules/.bin/ember --version
ember-cli: 4.7.0
node: 14.19.0
os: darwin x64
✨  Done in 2.64s.

~/repos/ember-template-lint-recreation/recreate-ember-template-lint--issue main*
❯ yarn run ember-template-lint --version
yarn run v1.22.19
$ /Users/jbremer/repos/ember-template-lint-recreation/recreate-ember-template-lint--issue/node_modules/.bin/ember-template-lint --version
4.15.0
✨  Done in 0.90s.
```
